skunkworks_crow/src/main/java/org/odk/share/views/ui/bluetooth/BtReceiverActivity.java

```diff
@@ -311,4 +320,24 @@ protected void onDestroy() {
         super.onDestroy();
         unregisterReceiver(bluetoothReceiver);
     }
+
+    @Override
+    public void onBackPressed() {
+        if (BluetoothUtils.isBluetoothEnabled()) {
+            new AlertDialog.Builder(this)
+                    .setTitle(getString(R.string.disable_bluetooth))
+                    .setMessage(getString(R.string.disable_bluetooth_msg))
+                    .setPositiveButton(android.R.string.yes, (DialogInterface dialog, int which) -> {
+                        receiverService.cancel();
+                        BluetoothUtils.disableBluetooth();
+                        super.onBackPressed();
+                    })
+                    .setNegativeButton(android.R.string.no, (DialogInterface dialog, int which) -> {
+                        receiverService.cancel();
```

Comment:
If the user presses `Cancel` button then why're we canceling the receiver service and navigating back the user to the main screen. Ideally, only dialog box should be dismissed.